src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/ServiceValueResolver.php

```diff
@@ -39,9 +39,15 @@ public function supports(Request $request, ArgumentMetadata $argument)
 
         if (\is_array($controller) && \is_callable($controller, true) && \is_string($controller[0])) {
             $controller = $controller[0].'::'.$controller[1];
+        } elseif (!\is_string($controller) || '' === $controller) {
+            return false;
         }
 
-        return \is_string($controller) && $this->container->has($controller) && $this->container->get($controller)->has($argument->getName());
+        if ('\\' === $controller[0]) {
```

Comment:
Do we really need to test first and then trim? Why not always trim?

Reply:
Trim means triggering copy-on-write when the controller string usually comes from shared memory, which means also recomputing the hash of the string, even if it doesn't change.